app/src/services/census/CensusStreamService.ts

```diff
@@ -96,9 +123,31 @@ export default class CensusStreamService implements ServiceInterface {
                     world_id: String(World.MILLER),
                 });
                 /* eslint-enable */
-                this.wsClient.emit(Events.PS2_META_EVENT, event);
+                // this.wsClient.emit(Events.PS2_META_EVENT, event);
                 CensusStreamService.logger.debug('Emitted Metagame Start event');
             }
         });
     }
+
+    private startMessageTimer(): void {
+        CensusStreamService.logger.info('Census message timer started');
+
+        this.messageTimer = setInterval(() => {
+            CensusStreamService.logger.debug('Census message timeout check running...');
+
+            this.messageThresholds.forEach((thresholdLimit, eventType) => {
+                const lastTime: Date | undefined = this.lastMessagesMap.get(eventType);
+                const threshold: number = new Date().getTime() - thresholdLimit;
+
+                if (!lastTime) {
+                    throw new ApplicationException('Undefined lastTime map entry, shouldn\'t be possible! Check constructor.');
+                }
+
+                if (lastTime.getTime() < threshold) {
+                    CensusStreamService.logger.error(`No Census messages received for event type "${eventType}" within expected threshold of ${thresholdLimit / 1000} seconds. Rebooting Connection.`);
+                    void this.wsClient.watch();
```

Comment:
This should be replaced with a resubscribe function or use the force flag. 

@microwavekonijn hurry up man! :P

Reply:
Heh, yeah once you've finished it, whatever supported reconnection method is available will be used.